types/Actions.d.ts

```diff
@@ -0,0 +1,134 @@
+declare module 'grafx-studio-actions' {
+
+    export type EditorEventType = 'layoutChanged' | 'frameTransformChanged' | 'variableValueChanged' | 'frameVisibilityChanged';
+
+    export interface ActionEventData {
+        type: EditorEventType;
+        source: FrameProperties|Variable|AlternateLayout;
+    }
+
+    /**
+     * An interface representing the API for working with actions.
+     * Provides access to objects for manipulating frames, variables,
+     * layouts, and document properties.
+     */
+    export interface ActionApi {
+        /** An object with methods for manipulating and accessing frame data */
+        frames: FramesController;
+        /** An object providing methods for getting and setting variable data */
+        variables: VariablesController;
+        /** An object for selecting a layout */
+        layouts: LayoutsController;
+        /** An object for retrieving the document size */
+        document: DocumentController;
+    }
+
+    interface FrameProperties {
+        readonly frameId: string;
+        readonly layoutId: string;
+        readonly x: number;
+        readonly y: number;
+        readonly width: number;
+        readonly height: number;
+        readonly rotationDegrees: number;
+        readonly rotationOriginY: number;
+        readonly scaleX: number;
+        readonly scaleY: number;
+        readonly included: boolean;
+        readonly fitMode: FitMode;
+        readonly minCopyfitting: number;
+        readonly maxCopyfitting: number;
+        readonly enableCopyfitting: boolean;
+        readonly rotationAngle: number;
+    }
+
+    interface PageProperties {
+        pageId: string;
+        layoutId: string;
+        width: number;
+        height: number;
+    }
+
+    enum FitMode {
+        fill,
+        fit,
+    }
+
+    interface AlternateLayout{
```

Comment:
This misses some properties, but it will use the inherited layout instead of the one saved in the document in a later version.

Reply:
@Matthiee I skipped a couple of properties on purpose, and tried to stick to the basics. Do you think we can start with the list in the current PR?

Reply:
@pietervp sounds good